docs/source/algorithms/SaveGDA-v1.rst

```diff
@@ -0,0 +1,118 @@
+  (normalised by TOF) in the bank
+
+The data then follows. We have 4 points per line, where each point
+consists of three values, ``TOF * 32`` (time-of-flight is scaled by 32
+for legacy reasons in MAUD - see :ref:`save_gda_tof` for a detailed
+explanation of where the TOF values come from), ``intensity * 1000``
+and ``error * 1000``. ``TOF``, ``intensity`` and ``error`` correspond
+to the ``x``, ``y`` and ``e`` columns of a Mantid workspaces
+respectively.
+
+All numbers apart from resolution are saved as integers. In addition,
+every line, including the header, must be exactly 80 characters long,
+so any short lines are right-padded with spaces.
+
+.. _save_gda_tof:
+
+D to TOF Conversion
+-------------------
+
+.. warning::
+
+   TOF values in the output file will only match the actual recorded
+   TOF values if the GSAS calibration file contains the correct
+   conversion factors for each bank.
+
+SaveGDA takes input in D-spacing, and applies the GSAS conversion
+(explained at :ref:`AlignDetectors <algm-AlignDetectors>`), using
+parameters from the calibration file, to convert back to
+time-of-flight. The caveat here is that, if the calibration file
+contains the wrong conversion factors, then the TOF values will not
+match the ones that were actually recorded.
+
+This is not necessarily a problem, as once the file is loaded into
+MAUD, as long as the same conversion factors are used (ie the MAUD
+calibration file should be created from the same GSAS calibration file
+as was used to run SaveGDA), the data will still be aligned in
+MAUD. When doing texture focusing with :ref:`ISIS_Powder for GEM
+<isis-powder-diffraction-gem-ref>`, matching up all the calibration
+files should be taken care of automatically.
+
+This approach is taken because it was impractical to create a MAUD
+calibration file containing the correct conversion factors for a
+workspace with a large number of banks - we just don't have enough
+data to do it. Instead we fake the time-of-flight recordings in order
+to get good alignment in MAUD.
+
+Usage
+-----
+
+.. testcode:: SaveGDA
+
+   import os
+
+   # Banks 1 to 4 of a previous texture focus in isis_powder
+   # We don't use the full 160 banks as the test becomes too slow
+   input_group = Load("GEM61785_D_texture_banks_1_to_4.nxs")
+
+   output_file = os.path.join(config["defaultsave.directory"], "GEM61785.gda")
+   SaveGDA(InputWorkspace=input_group,
+           Filename=output_file,
+	   GSASParamFile="GEM_PF1_PROFILE.IPF",
+           # Assign spectra 1, 2 and 3 to bank 2 in calib file, and spectrum 4 to bank 3
+	   GroupingScheme=[2, 2, 2, 3])
+
+   with open(output_file) as f:
+       file_contents = f.read().split("\n")
+
+   # Print the header and the 4 lines from the middle of the file
+   # rstrip the header just to make the doctest script happy
+   print(file_contents[0].rstrip())
+   for i in range(100, 104):
+       print(file_contents[i])
+
+.. testcleanup:: SaveGDA
+
+   os.remove(output_file)
```

Comment:
You'll want to remove the ws from the ADS too, otherwise it starts to fill as each doc test finishes